### PR TITLE
Feature/workallocation 7

### PR DESCRIPTION
**Pull request by udaydenduluri33**
### JIRA link (if applicable) ###

EUI-3602, https://tools.hmcts.net/jira/browse/EUI-3382, https://tools.hmcts.net/jira/browse/EUI-3470, https://tools.hmcts.net/jira/browse/EUI-3428, https://tools.hmcts.net/jira/browse/EUI-3437, etc

### Change description ###

Several Bug fixes which are separate PRs for Work Allocation

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: api/workAllocation/index.ts [hunk 1]
```diff
@@ -63,8 +63,7 @@ export async function searchTask(req: EnhancedRequest, res: Response, next: Next
     res.status(status);
     // Assign actions to the tasks on the data from the API.
     if (data) {
-      const caseworkers: Caseworker[] = await retrieveAllCaseWorkers(req, res);
-      assignActionsToTasks(data.tasks, req.body.view, caseworkers);
+      assignActionsToTasks(data.tasks, req.body.view, req.session.caseworkers);
```
Comment (paul-graham) [inline api/workAllocation/index.ts:66]: I already requested this be changed on https://github.com/hmcts/rpx-xui-webapp/pull/953. Why was this merged? 🤯

**Review by paul-graham — CHANGES_REQUESTED** [review]
I've not looked at the rest of this PR but there was an issue with the very first file. This shouldn't have been merged.

File: api/workAllocation/util.ts [hunk 2]
```diff
@@ -88,24 +90,45 @@ function getAssigneeName(task: any, caseworkers: Caseworker[]): string {
   return null
 }
 
-export function mapCaseworkerData(caseWorkerData: any[]): any[] {
+export function mapCaseworkerData(caseWorkerData: CaseworkerApi[]): Caseworker[] {
+  const caseworkers: Caseworker[] = []
```
Comment (kuda1992) [inline api/workAllocation/util.ts:94]: Just a suggestion here. We could use a map `caseWorkerData.map(caseWork => ({...mappedObject}))`

File: api/workAllocation/index.ts (hunk 1 above)
Comment (udaydenduluri33) [reply api/workAllocation/index.ts:66]: This has been merged after discussing with @ronaldmansveld  that this is not an issue. I can still explain you if you have the time. Thanks.

**Review by kuda1992 — COMMENTED** [review]
Looks good just one minor change requested

File: api/workAllocation/util.ts (hunk 2 above)
Comment (udaydenduluri33) [reply api/workAllocation/util.ts:94]: Agreed. I believe it is nice to have

**Review by kuda1992 — APPROVED** [review]

File: api/workAllocation/index.ts (hunk 1 above)
Comment (udaydenduluri33) [reply api/workAllocation/index.ts:66]: As per call This is agreed to be revisited. Can you approve it now? Thanks

**Review by paul-graham — APPROVED** [review]
Some long lines of code in there but it just about passes linting. They don't look very pretty but that's a purely stylistic thing.

As @udaydenduluri33 has mentioned, it was agreed on a call that the change made in `api/workAllocation/index.ts` needs to be fixed because there are scenarios where it will result in a failure. The agreement was that this should be addressed within the next 24 hours.

With that in mind, and with agreement from @ronaldmansveld and @Mo-Lala84, I'll approve this PR.

**Comment by paul-graham** [comment]
https://tools.hmcts.net/jira/browse/EUI-3611 has been raised to deal with the issue surrounding caseworkers not having been loaded yet.

File: api/workAllocation/index.ts (hunk 1 above)
Comment (paul-graham) [reply api/workAllocation/index.ts:66]: https://tools.hmcts.net/jira/browse/EUI-3611

**Pull request merged**